### ISSUE TITLE
Static port for captures

### ISSUE THREAD

**Issue by auras** [malfunkt/hyperfox#16]
I want to run hyperfox inside a docker container. But I cannot configure docker to forward a port without knowing it beforehand and for that I would need to be able to configure the port to view the captures.

**Comment by elico** [comment]
@auras have you tried using bridged interface for the docker machine?
This way you would be able to access the docker directly without any proxy\forwarding.

**Comment by auras** [comment]
Yeah, I know I can do that. But I need something predictable that can be
shared with a whole team.

**Comment by elico** [comment]
@auras So what do you need?
Since it's a docker the inside port will always be the same....
Do you need to register in some place a docker instance external(ie host) port?
Do you need to customize it or publish it?

**Comment by auras** [comment]
Both. To expose the port I would need to know it in advance.

**Comment by elico** [comment]
@auras The internal would be setup by you and only you as the docker maintainer.
There are couple variables which you can use to do that but I am still not sure what and how.
The settings of hyperfox inside the docker container can be set statically for example 18080.
Then you can play with each instance when you spin it up to set the desired external port.
There shouldn't be ever a situation that you spin a docker and you give it "auto internal port choice".. it's always based on some settings.. so only left the external.

**Comment by auras** [comment]
I've never used go but in service.go line 330, doesn't that mean it's
looking for a random open port?


	// Looking for a port to listen to.
	ln, err := net.Listen("tcp", serviceBindHost+":0")

**Comment by elico** [comment]
@auras I have not seen that.
Well the code needs some changing.
There is a flag: defaultPort 
But if it's not used then it's either and error or maybe I am not understanding the code right.
@pulkitsharma07 @xiam , Can you clear it out?
Should the flag be used? Is it used?

**Comment by pulkitsharma07** [comment]
@auras , yes you are right, It will be a random port (https://golang.org/pkg/net/#Listen, `If the port in the address parameter is empty or "0", as in "127.0.0.1:" or "[::1]:0", a port number is automatically chosen`). There is no functionality (it seems) to have it listen on a specific port. I am not familiar with the code base, So, I guess you'll need to give a PR to have this functionality. If you need it at this moment, you can always change that line (or add a flag) to set it have it listen to a specific port. Another workaround, is to close all other ports in docker container (So, that we can make that port somewhat deterministic), I am not sure how to achieve this, Just my two cents here.

@elico , That flag is for the port the proxy should bind to. I am not sure what @auras 's use case is. `service.go` mentioned by @auras is concerned with other services i.e. things other than the actual proxy.

The proxy default port is mentioned here https://github.com/malfunkt/hyperfox/blob/71d50a7d7325b19cba86bf2112370c8ea44bcbaf/main.go#L41 and as I mentioned above, you can change it using the `p` flag.

**Comment by auras** [comment]
I get that and I know how I would go about to make it static. I wouldn't
have created this issue if I worked with go before. But I didn't and
couldn't manage to download the dependencies and build the project.

**Comment by pulkitsharma07** [comment]
Cool, I'll create a PR later.

**Comment by auras** [comment]
Thank you

**Comment by elico** [comment]
@auras What OS are you using?

**Comment by auras** [comment]
I'm building on a ubuntu 16.04 docker container

**Comment by elico** [comment]
@auras @pulkitsharma07 I tried to build it for both Linux and FreeBSD but I only managed to build hyperfox for linux 64 bit because of something with sqlite
```
 github.com/mattn/go-sqlite3
../../go/src/github.com/mattn/go-sqlite3/sqlite3_go18.go:18:10: undefined: SQLiteConn
```

**Comment by auras** [comment]
Which version of go did you use?

**Comment by elico** [comment]
@auras https://github.com/malfunkt/hyperfox/tree/v1.9.8
